src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -278,6 +280,315 @@ struct mac_addr_list
   }
 };
 
+struct aim_counter
+{
+  using result_type = query::aim_counter::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type key, const boost::any& arg1)
+  {
+    const auto dbgIpData = boost::any_cast<query::aim_counter::debug_ip_data_type>(arg1);
+    auto pdev = get_pcidev(device);
+
+    // read counter values
+    std::string aimName("aximm_mon_");
+    aimName = aimName + std::to_string(dbgIpData->m_base_address);
+
+    std::string namePath = pdev->get_sysfs_path(aimName.c_str(), "name");
+
+    std::size_t pos = namePath.find_last_of('/');
+    std::string path = namePath.substr(0, pos+1);
+    path += "counters";
+
+    result_type retvalBuf(9, 0);
+
+    std::ifstream ifs(path.c_str());
+    if(!ifs) {
+      std::cout << "\nINFO: Incomplete AIM counter data in " << path << std::endl;
+      return retvalBuf;
+    }
+
+    const size_t sz = 256;
```

Comment:
Make sure value is done at compile time (it may be for 'const' but is not guaranteed)
`constexpr size_t sz = 256`